torch/csrc/jit/passes/python_print.cpp

```diff
@@ -1104,10 +1114,20 @@ struct PythonPrintPass {
   }
 
   void printClass(const c10::SerializableTypePtr& classType) {
-    body_ << "class " << classType->basename() << ":\n";
+    body_ << "class " << classType->basename();
```

Comment:
I think it would be fine here to simply have another switch and then use helper functions to emit the body. The problem with the current way is that it won't support interfaces anyway, so we'd need to introduce the switch statement still, and the parser part already does it with a switch. So it is a bit confusing that input is doing via switches but output is done via virtual methods. I like the idea of making NamedTuple/Class/Interface all inherit from NamedTypePtr, which could then be what goes in the compilation units type table.